cmd/talosctl/cmd/mgmt/gen/secureboot.go

```diff
@@ -0,0 +1,228 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package gen
+
+import (
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/foxboron/go-uefi/efi"
+	"github.com/foxboron/go-uefi/efi/signature"
+	"github.com/foxboron/go-uefi/efi/util"
+	"github.com/google/uuid"
+	"github.com/siderolabs/crypto/x509"
+	"github.com/spf13/cobra"
+
+	"github.com/siderolabs/talos/pkg/machinery/config/generate/secrets"
+)
+
+var genSecurebootCmdFlags struct {
+	outputDirectory string
+}
+
+// genSecurebootCmd represents the `gen secureboot` command.
+var genSecurebootCmd = &cobra.Command{
+	Use:   "secureboot",
+	Short: "Generates secrets for the SecureBoot process",
+	Long:  ``,
+}
+
+var genSecurebootUKICmdFlags struct {
+	commonName string
+}
+
+// genSecurebootUKICmd represents the `gen secureboot uki` command.
+var genSecurebootUKICmd = &cobra.Command{
+	Use:   "uki",
+	Short: "Generates a certificate which is used to sign boot assets (UKI)",
+	Long:  ``,
+	Args:  cobra.NoArgs,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		return generateSigningCerts(genSecurebootCmdFlags.outputDirectory, "uki", genSecurebootUKICmdFlags.commonName)
+	},
+}
+
+var genSecurebootPCRCmdFlags struct {
+	commonName string
+}
+
+// genSecurebootPCRCmd represents the `gen secureboot pcr` command.
+var genSecurebootPCRCmd = &cobra.Command{
+	Use:   "pcr",
+	Short: "Generates a certificate which is used to sign TPM PCR values",
+	Long:  ``,
+	Args:  cobra.NoArgs,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		return generateSigningCerts(genSecurebootCmdFlags.outputDirectory, "pcr", genSecurebootPCRCmdFlags.commonName)
+	},
+}
+
+var genSecurebootDatabaseCmdFlags struct {
+	enrolledCertificatePath                string
+	signingCertificatePath, signingKeyPath string
+}
+
+// genSecurebootDatabaseCmd represents the `gen secureboot database` command.
+var genSecurebootDatabaseCmd = &cobra.Command{
+	Use:   "database",
+	Short: "Generates a UEFI database to enroll the signing certificate",
+	Long:  ``,
+	Args:  cobra.NoArgs,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		return generateSecureBootDabase(
+			genSecurebootCmdFlags.outputDirectory,
+			genSecurebootDatabaseCmdFlags.enrolledCertificatePath,
+			genSecurebootDatabaseCmdFlags.signingKeyPath,
+			genSecurebootDatabaseCmdFlags.signingCertificatePath,
+		)
+	},
+}
+
+func checkedWrite(path string, data []byte, perm fs.FileMode) error { //nolint:unparam
+	if err := validateFileExists(path); err != nil {
+		return err
+	}
+
+	if dirname := filepath.Dir(path); dirname != "." {
+		if err := os.MkdirAll(dirname, 0o700); err != nil {
+			return err
+		}
+	}
+
+	fmt.Fprintf(os.Stderr, "writing %s\n", path)
+
+	return os.WriteFile(path, data, perm)
+}
+
+func generateSigningCerts(path, prefix, commonName string) error {
```

Comment:
we need to also pass in the key bits, pcr certs needs a 2048 key (tpm doesn't allow loading an rsa key with size > 2048)

Reply:
fixed, thanks!